primitives/src/assets.rs

```diff
@@ -476,4 +481,27 @@ where
         }
         .map_err(FungibleLedgerError::InvalidTransfer)
     }
+
+    #[inline]
+    fn burn(
+        asset_id: AssetId,
+        who: &C::AccountId,
+        amount: Balance,
+    ) -> Result<(), FungibleLedgerError> {
+        Self::ensure_valid(asset_id)?;
+        Self::can_withdraw(asset_id, who, amount)?;
+        if asset_id == A::NativeAssetId::get() {
+            <Native as Currency<C::AccountId>>::withdraw(
```

Comment:
For native assets, it's transferred instead of burned, kind of confused to me. Perhaps leaving some comments here would make more sense.

Reply:
It is not transferred, the funciton is `withdraw`. The `Transfer` parameter are `reasons` which are used for `frozen()` checks